### PR TITLE
fix: gs quick wizard table header index

### LINKED ISSUE #17974: [Bug]: Internal server error is displayed when the user selects "Sheets" as datasource from Quick Page Wizard
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When the user selects google sheets as datasources from the 'Quick Page Wizard' page internal server error is displayed.

### Steps To Reproduce

1. Go to this url:- https://release.app.appsmith.com/applications
2. Click on **New** button and create your workspace
3. Click on **Generate page from data table**
4. Now on **Quick Page Wizard** select **Sheets** as your datasource from the dropdown
5. Observe the error 

**Actual Result**
Internal server error is observed when **"Sheets"** as datasource is selected.

**Expected Result**
User should be able to select **"Sheets"** as datasource without any error.

**Note**
Please refer to the attached video:- https://jiju8jbmwa.vmaker.com/record/uQ4egV0MQPmIK7Jp 

### Public Sample App

_No response_

### Version

Release

Front logo [Front conversations](https://app.frontapp.com/open/top_61x1o)

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

Table header index passed to successfully make quick wizard work

Fixes #17974 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/MethodConfig.java [hunk 1]
```diff
@@ -70,8 +70,9 @@ public MethodConfig(Map<String, Object> formData) {
         this.sheetName = getTrimmedStringDataValueSafelyFromFormData(formData, SHEET_NAME);
         this.rowObjects = getTrimmedStringDataValueSafelyFromFormData(formData, FieldName.ROW_OBJECTS);
 
-        if (validDataConfigurationPresentInFormData(formData, FieldName.WHERE, new TypeReference<Map<String, Object>>() {
-        })) {
+        if (validDataConfigurationPresentInFormData(formData, FieldName.WHERE,
+                new TypeReference<Map<String, Object>>() {
+                })) {
             Map<String, Object> whereForm = getDataValueSafelyFromFormData(
```
Comment (vaibh1297) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/MethodConfig.java:76]: There is no code change here. Please undo it, as this is creating confusion.

**Review by Irongade — APPROVED** [review]

**Pull request merged**